processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
Given that I don't think that moving the `indexOf` is really buying us anything.  It's either code in `BufferedIndexed` or it's up here.  Why not just move it back to where it was and leave it be.